packages/files/file.pony

```diff
@@ -478,28 +477,28 @@ class File
     let writev_batch_size = @pony_os_writev_max()
     while pending_total > 0 do
       // Determine the number of bytes and buffers to send.
-      num_to_send = (_pending_writev.size().i32() / 2) - num_sent.i32()
+      num_to_send = _pending_writev.size().i32() - num_sent.i32()
```

Comment:
Before I merge, can you explain the removal of `/2` here and `*2` etc elsewhere?

Reply:
Previously, each io vector in the array took up two indices, one for the pointer itself and one for the size.

So, for example, if there were 4 io vectors, `_pending_writev: Array[USize]` would be 8 elements long, with indices 1, 3, 5, & 7 being pointers (converted to `USize`), and indices 2, 4, 6, & 8 being the associated sizes to those pointers.

After this change, if there are 4 io vectors, then `_pending_writev: Array[(Pointer[U8] tag, USize)]` will be 4 elements long, with each element being a tuple tuple containing both parts that were previously encoded at separate indices in the array.